### ISSUE TITLE
Add a reset() method for view blocks

### ISSUE THREAD

**Issue by burzum** [cakephp/cakephp#7561]
I had the idea to create reuseable widgets because we want tables to be the same everywhere in your app and be able to simply change the markup as needed without go through all views.

So the idea was to figure out if we can view blocks for this like this:

```
start table
    start table-header
    end
    start table-body
    end
    start table-footer
    end
end
```

So far so good, this would work, but the problem starts when you want to add the next table. There is no way to reset the view blocks. You'll keep adding content to the existing blocks.

I think adding a reset() method would resolve that problem and give view blocks even more power by being able to create reuseable widgets that use view blocks.

**Comment by markstory** [comment]
You can overwrite/reset the content of a block using `View->assign('table', '')`.

**Comment by jeffblack360** [comment]
@markstory I've added a 'reset' method to src/View/View.php that is a convenience and merely a wrapper that calls assign('table,''); along with accompanying unit-tests. I this OK for a PR?

**Comment by dereuromark** [comment]
@jeffblack360 Just make the PR if you got it all anyway :) We can then easier discuss it.
I like the wrapper method as a cleanly speaking API function better (vs the hacky assign solution otherwise).

**Comment by markstory** [comment]
@jeffblack360 A pull request would be great.

**Comment by jeffblack360** [comment]
@dereuromark @markstory Will do!

**Comment by jeffblack360** [comment]
@markstory @dereuromark What is the best way to mentiion this in the documentation and coordinate it with the release the feature is carried in?

**Comment by markstory** [comment]
@jeffblack360 There is a `3.2` branch in the book. Doing a pull request against that branch will ensure that the docs are only published when 3.2 hits an RC. We use `.. versionadded:: 3.2` directives in the book to indicate when features were added.

**Comment by jeffblack360** [comment]
@markstory Thank you sir! I'll update the appropriate doc branch accordingly and submit a PR.
